lib/std/mem.zig

```diff
@@ -273,6 +273,33 @@ pub fn set(comptime T: type, dest: []T, value: T) void {
         d.* = value;
 }
 
+/// Zero initializes the type.
+/// This can be used to zero initialize a C-struct.
+pub fn zeroes(comptime T: type) T {
+    if (@sizeOf(T) == 0) return T{};
+
+    if (comptime meta.containerLayout(T) != .Extern) {
```

Comment:
Should allow packed as well?

Reply:
not necessary for merge, made complicated by #3133